### PR TITLE
Add support for pre-supplying host reference bowtie2 index

### PR DESCRIPTION
**Pull request by jfy133**
To close https://github.com/nf-core/mag/issues/452

(I would rather replace this with the official nf-=core bowtie2 module, but it's not an easy swap out, so will do that later)

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mag/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/mag _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by maxibor — APPROVED** [review]
LGTM. 
IGenome is not always convenient anyways :)

File: modules/local/bowtie2_removal_build.nf [hunk 1]
```diff
@@ -10,14 +10,14 @@ process BOWTIE2_REMOVAL_BUILD {
...
-    path "versions.yml"   , emit: versions
+    path "*.bt2"        , emit: index
+    path "versions.yml" , emit: versions
 
     script:
     def args = task.ext.args ?: ''
     """
     mkdir bowtie
-    bowtie2-build --threads $task.cpus $fasta "bt2_index_base"
+    bowtie2-build --threads $task.cpus $fasta ${fasta.simpleName}
```
Comment (danilodileo) [inline modules/local/bowtie2_removal_build.nf:20]: Just my curiosity: does simpleName apply to channels as well?

Comment (jfy133) [reply modules/local/bowtie2_removal_build.nf:20]: You can use it on any file object: https://www.nextflow.io/docs/latest/script.html#getting-file-attributes

Comment (jfy133) [reply modules/local/bowtie2_removal_build.nf:20]: I'm going to merge in now, but if you have any other issues please shout

**Pull request merged**